src/Symfony/Component/PropertyInfo/PropertyTypeExtractorInterface.php

```diff
@@ -21,7 +24,17 @@ interface PropertyTypeExtractorInterface
     /**
      * Gets types of a property.
      *
-     * @return Type[]|null
+     * @deprecated since Symfony 7.1, use "getType" instead.
+     *
+     * @return LegacyType[]|null
      */
     public function getTypes(string $class, string $property, array $context = []): ?array;
+
+    /**
+     * Gets type of a property.
+     *
+     * @param class-string         $class
+     * @param array<string, mixed> $context
+     */
+    public function getType(string $class, string $property, array $context = []): ?Type;
```

Comment:
This is a BC break as hinted by the failing CI build, the method needs to be virtual for now
```php
* @method ?Type getType(string $class, string $property, array $context = []) 
```
(I intentionally removed the description as it's stating the obvious)
We'll add really add it in when removing `getTypes()` in 8.0.